src/scripts/party/evolutions/TimedLevelEvolution.ts

```diff
@@ -0,0 +1,32 @@
+class TimedLevelEvolution extends LevelEvolution {
+
+    startHour: number; // including
+    endHour: number; // excluding
+
+    constructor(basePokemon: string, evolution: string, level: number, startHour: number, endHour: number) {
+        super(basePokemon, evolution, level);
+        this.startHour = startHour;
+        this.endHour = endHour;
+    }
+
+    isSatisfied(): boolean {
+        const currentHour = new Date().getHours();
+        // Check that evolution is within reached regions
+        return PokemonHelper.calcNativeRegion(this.evolvedPokemon) <= player.highestRegion()
+        // Check current time within evolution hours
+        && currentHour >= this.startHour && currentHour < this.endHour
```

Comment:
I think we need different logic here to handle NightEvolution, there is no `currentHour` which can be both greater than 18 and less then 6.

Maybe when constructing the TimedEvolutions, if startHour is greater than endHour, we swap them and set a flag to invert the check, so we are effectively checking the current hour is not in the complementary range.